internal/handler/handler.go

```diff
@@ -86,48 +114,39 @@ func (svc userMgmtSvc) SignUp(w http.ResponseWriter, r *http.Request) {
 }
 func (svc userMgmtSvc) Login(w http.ResponseWriter, r *http.Request) {
 	var credential model.LoginCredentials
-	bytes, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		log.Error(err)
-		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrReadingReqBody), nil)
-		return
-	}
-	err = json.Unmarshal(bytes, &credential)
+	status, err := request.FromJson(r, &credential)
 	if err != nil {
 		log.Error(err)
-		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrUnmarshall), nil)
+		response.ToJson(w, status, err.Error(), nil)
 		return
 	}
-	err = validator.Validate(credential)
+	err = validatePass(credential.Password)
 	if err != nil {
 		log.Error(err)
-		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrValidate), nil)
+		if err.Error() != codes.GetErr(codes.ErrPassRegex) {
+			response.ToJson(w, http.StatusBadRequest, err.Error(), nil)
+			return
+		}
+		response.ToJson(w, http.StatusInternalServerError, err.Error(), nil)
 		return
 	}
 	resp := svc.logic.Login(w, credential)
 	response.ToJson(w, resp.Status, resp.Message, resp.Data)
-
 }
 func (svc userMgmtSvc) Activation(w http.ResponseWriter, r *http.Request) {
-	var data map[string]string
-	bytes, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		log.Error(err)
-		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrReadingReqBody), nil)
-		return
-	}
-	err = json.Unmarshal(bytes, &data)
+	var data model.Activate
+	status, err := request.FromJson(r, &data)
 	if err != nil {
 		log.Error(err)
-		response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrUnmarshall), nil)
+		response.ToJson(w, status, err.Error(), nil)
 		return
 	}
-	resp := svc.logic.Activate(data["user_id"])
+	resp := svc.logic.Activate(data.UserId)
 	response.ToJson(w, resp.Status, resp.Message, resp.Data)
 }
 
 func (svc userMgmtSvc) UserDetails(w http.ResponseWriter, r *http.Request) {
 	id := session.GetSession(r.Context())
-	resp := svc.logic.UserData(id)
+	resp := svc.logic.UserData(id.(string))
```

Comment:
comma, ok idiom was to be used to prevent panics.

Reply:
done